crates/rocksdb/src/rock_access.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2024 -  Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use rocksdb::perf::MemoryUsageBuilder;
+use rocksdb::ColumnFamilyDescriptor;
+
+use crate::CfName;
+use crate::DbSpec;
+use crate::RocksError;
+
+pub trait RocksAccess {
+    fn open_db(db_spec: &DbSpec<Self>) -> Result<Self, RocksError>
+    where
+        Self: Sized;
+
+    fn flush_memtables_atomically(&self, cfs: &[&CfName], wait: bool) -> Result<(), RocksError>;
+    fn flush_wal(&self, sync: bool) -> Result<(), RocksError>;
+    fn cancel_all_background_work(&self, wait: bool);
+    fn set_options_cf(&self, cf: &CfName, opts: &[(&str, &str)]) -> Result<(), RocksError>;
+    fn get_property_int_cf(&self, cf: &CfName, property: &str) -> Result<Option<u64>, RocksError>;
+    fn record_memory_stats(&self, builder: &mut MemoryUsageBuilder);
+}
+
+impl RocksAccess for rocksdb::DB {
+    fn open_db(db_spec: &DbSpec<Self>) -> Result<Self, RocksError> {
+        let column_families = db_spec
+            .column_families
+            .iter()
+            .map(|(name, opts)| ColumnFamilyDescriptor::new(name.clone(), opts.clone()));
+        rocksdb::DB::open_cf_descriptors(&db_spec.db_options, &db_spec.path, column_families)
+            .map_err(RocksError::from_rocksdb_error)
+    }
+
+    fn flush_memtables_atomically(&self, cfs: &[&CfName], wait: bool) -> Result<(), RocksError> {
```

Comment:
Wondering whether the suffix `atomically` could be confusing because doesn't it depend on the configured `rocksdb::Options`, with which the db was opened, whether the flush is atomic or not.

Reply:
Good call. Maybe just `flush_memtables` is enough.